cmd/sync.go

```
@@ -115,6 +131,19 @@ func (raw *rawSyncCmdArgs) parsePatterns(pattern string) (cookedPatterns []strin
 	return
 }
 
+func (raw *rawSyncCmdArgs) parseCFDMode() common.CFDMode {
+	cfdMode := strings.ToLower(raw.cfdMode)
+	if cfdMode == strings.ToLower(common.CFDModeFlags.TargetCompare().String()) {
+		return common.CFDModeFlags.TargetCompare()
+	} else if cfdMode == strings.ToLower(common.CFDModeFlags.Ctime().String()) {
+		return common.CFDModeFlags.Ctime()
+	} else if cfdMode == strings.ToLower(common.CFDModeFlags.CtimeMtime().String()) {
+		return common.CFDModeFlags.CtimeMtime()
+	} else {
+		return common.CFDModeFlags.TargetCompare()
```

let's not allow any arbitrary string but instead let's fail if user provides an invalid string.
To support the case where user does not provide the option and we want to default to TargetCompare, check for empty string explicitly.